prow/test/integration/test/tide_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/github"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestTide(t *testing.T) {
+	const (
+		baseRef = "master"
+		// matches tide deployment configurating in //prow/test/integration/prow/config.yaml
+		// for tide to query the right org/repo
+		org  = "fake-org-tide"
+		repo = "fake-repo-tide"
+	)
+
+	clusterContext := getClusterContext()
+	t.Logf("Creating client for cluster: %s", clusterContext)
+
+	kubeClient, err := NewClients("", clusterContext)
+	if err != nil {
+		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+	}
+
+	githubClient := github.NewClient(func() []byte { return nil }, func(b []byte) []byte { return b }, "", "http://localhost/fakeghserver")
+
+	t.Run("merge single job", func(t *testing.T) {
+		ctx := context.Background()
+		title := RandomString(t)
+		sha := RandomString(t)
+		baseSHA := RandomString(t)
+
+		prowJob, err := createPR(ctx, githubClient, kubeClient, org, repo, title, sha, baseSHA, baseRef)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Cleanup(func() {
+			if err := kubeClient.Delete(ctx, prowJob); err != nil {
```

Comment:
should we also delete the created PR? I'm trying to think the scenario of incremental testing on existing cluster